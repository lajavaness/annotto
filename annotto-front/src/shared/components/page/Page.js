import PropTypes from 'prop-types'
import React from 'react'

import * as Styled from 'shared/components/page/__styles__/Page.styles'

const Page = ({ id, children }) => <Styled.Root data-testid={`__${id}-page__`}>{children}</Styled.Root>

Page.propTypes = {
	/** Id of the page. */
	id: PropTypes.string,
	/** The content of the page. */
	children: PropTypes.node,
}

Page.defaultProps = {
	id: null,
	children: null,
}

export default Page
