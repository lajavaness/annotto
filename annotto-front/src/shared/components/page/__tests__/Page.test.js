import React from 'react'

import { render } from '@testing-library/react'

import Page from '../Page'

const getInstance = () => <Page />

describe('Root', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
