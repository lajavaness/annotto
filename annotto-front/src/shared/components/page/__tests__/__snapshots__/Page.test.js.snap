// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-bdfBQB bMrpEf"
    data-testid="__null-page__"
  />
</DocumentFragment>
`;
